optax/_src/loss.py

```diff
@@ -461,6 +461,35 @@ def ctc_loss(logits: chex.Array,
   return per_seq_loss
 
 
+def convex_kl_divergence(log_predictions: chex.Array,
+                  targets: chex.Array) -> chex.Array:
+  """Computes a convex version of the Kullback-Leibler divergence loss.
+
+  Measures the information gain achieved if target probability distribution
+  would be used instead of predicted probability distribution.
+  This version is jointly convex in p (targets) and q (log_predictions).
+
+  References:
+    [Kullback, Leibler, 1951](https://www.jstor.org/stable/2236703)
+
+  Args:
+    log_predictions: Probabilities of predicted distribution with shape
+      [..., dim]. Expected to be in the log-space to avoid underflow.
+    targets: Probabilities of target distribution with shape [..., dim].
+      Expected to be strictly positive.
+
+  Returns:
+    Kullback-Leibler divergence of predicted distribution from target
+    distribution with shape [...].
+  """
+  chex.assert_type([log_predictions, targets], float)
+  chex.assert_tree_all_finite(log_predictions)
```

Comment:
This value assertion will not work in unchexified code (see [the docs](https://github.com/deepmind/chex#static-and-value-aka-runtime-assertions)), so it's probably better to remove it from the default implementation.

Reply:
I've removed `chex.assert_tree_all_finite(...)`. Should I additionally remove `chex.assert_type(...)` as well? I see `chex.assert_type(...)` in every loss function.

Reply:
This is not necessary because `chex.assert_type` is a static assertions (i.e. it does not need access to actual tensor values).